File: 28-01-23/validate_BST/sol.java
class Solution {
    List<Integer> arr= new ArrayList<>();
    void inorder(TreeNode node){
            if(node==null){
                return ;
            }
            inorder(node.left);
            arr.add(node.val);
            inorder(node.right);
        }
        boolean isSorted(){
            for(int i=0;i<arr.size()-1;i++){
                if(arr.get(i)>=arr.get(i+1)){
                    return false;
                }
            }
            return true;
        }
    public boolean isValidBST(TreeNode root) {
        inorder(root);
        return isSorted();
    }
}

File: 28-01-23/validate_BST/sol.py
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        arr=[]
        def inorder(node):
            if not node:
                return
            inorder(node.left)
            arr.append(node.val)
            inorder(node.right)
        inorder(root)
        if (arr==sorted(arr) and len(set(arr))==len(arr)):
            return True
        else:
            return False
